### PR TITLE
Add deer mill passage data

### PR DESCRIPTION
**Pull request by liz-fw**
Ryan Revnak sent updated datasets for deer and mill containing data from 2021-2023 for inclusion in the EDI package.

This PR adds in updated `redd` data for `mill`, `holding` data for `deer`, and `upstream passage counts` for both `deer` and `mill` to the stream-specific qc markdowns, and also then incorporates them into the standard format datasets.

### CODE REVIEW

File: data-raw/qc-markdowns/adult-upstream-passage-monitoring/deer-creek/deer-creek-upstream-passage-monitoring.Rmd
Comment (alvizek) [inline data-raw/qc-markdowns/adult-upstream-passage-monitoring/deer-creek/deer-creek-upstream-passage-monitoring.Rmd:1]: My one question here is - are the pre-2021 and 2021-2023 data the same? It seems like the pre-2021 may be passage estimates whereas the 2021-2023 are counts.

File: data-raw/qc-markdowns/adult-upstream-passage-monitoring/mill-creek/mill-creek-upstream-passage-monitoring-qc-checklist.Rmd
Comment (alvizek) [inline data-raw/qc-markdowns/adult-upstream-passage-monitoring/mill-creek/mill-creek-upstream-passage-monitoring-qc-checklist.Rmd:1]: Same question as for Deer - counts v passage estimates

**Review by alvizek — COMMENTED** [review]
I did a quick review and my one flag is making sure we are combining compatible data. For Deer/Mill adult passage, I'm wondering if the old dataset was passage estimates and the new is just raw counts. I don't think we can combine the two. We would want them in separate tables. Let's connect on this when we meet.

**Comment by liz-fw** [comment]
modified to be pulling from the raw count tabs of yearly data files for both deer and mill

File: data-raw/qc-markdowns/adult-upstream-passage-monitoring/deer-creek/deer-creek-upstream-passage-monitoring.Rmd
Comment (alvizek) [reply data-raw/qc-markdowns/adult-upstream-passage-monitoring/deer-creek/deer-creek-upstream-passage-monitoring.Rmd:1]: I'm struggling with all the NAs in this dataset. I think NAs should be 0s. I see that you do this later in the code! 

Is the assumption that "salmon" are spring? I was wondering why the tab was called "spring".

File: data-raw/qc-markdowns/adult-upstream-passage-monitoring/deer-creek/deer-creek-upstream-passage-monitoring.Rmd [hunk 1]
```diff
@@ -39,63 +39,413 @@ Some missing data for variable flow
...
+```{r, message=FALSE, warning=FALSE}
+all_years_raw <- bind_rows(deer_2013, deer_2014, deer_2015, deer_2016,
+                           deer_2017, deer_2018, deer_2019, deer_2020,
+                           deer_2021, deer_2022) |> 
+  glimpse()
+
+all_deer_data <- all_years_raw |> 
+  select(-count_type) |> 
+  # pulled from excel tabs
+  mutate(viewing_adjust = case_when(is.na(viewing_adjust) | viewing_adjust == 0 ~ "VAKI",
```
Comment (alvizek) [inline data-raw/qc-markdowns/adult-upstream-passage-monitoring/deer-creek/deer-creek-upstream-passage-monitoring.Rmd:415]: how do you feel about all lower case? i know VAKI and DIDSON are abbrev but i like consistency in lower case.

File: data-raw/qc-markdowns/adult-upstream-passage-monitoring/mill-creek/mill-creek-upstream-passage-monitoring-qc-checklist.Rmd
Comment (alvizek) [reply data-raw/qc-markdowns/adult-upstream-passage-monitoring/mill-creek/mill-creek-upstream-passage-monitoring-qc-checklist.Rmd:1]: Same comments apply - I think we should transform NA counts to 0. Consider all lowercase for viewing_adjust.

**Review by alvizek — COMMENTED** [review]
This is pretty much there. Few notes -
- consider all lowercase for categorical variables
- make sure NA counts are transformed to 0 (let me know if want to talk about it)
- run - do we know run?
- update the metadata info in these markdowns to Ryan Revnak and maybe also include the contact info about the other guy (Doug?) just for reference.

Let me know when done and then I'll approve this.

**Comment by liz-fw** [comment]
@alvizek I made the updates listed above and should be good to go

**Review by alvizek — APPROVED** [review]
All changes implemented

**Pull request merged**